### PR TITLE
feat(yandex.ru): add presence

### PR DESCRIPTION
**Pull request by Nolo001-Aha**
## Description 

Add a presence for Yandex (a search engine popular in Russia and CIS countries). Presence shows the user's current search query/activity across the following services:

- Yandex Search
- Yandex Images
- Yandex Videos
- Yandex News
- Yandex Maps

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

Screenshot 1

Screenshot 2

Screenshot 3

</details>

### CODE REVIEW

**Comment by DarkVillager** [comment]
@Nolo001-Aha use variables for frequently accessed code

**Comment by RisingSunLight42** [comment]
Also, maybe use a switch statement instead of a if statement (in combination of the previous comment by Dark)

**Comment by Nolo001-Aha** [comment]
> Also, maybe use a switch statement instead of a if statement (in combination of the previous comment by Dark)

I think using if/else if is better in this situation because I use the **startsWith** method, and it returns a bool. I'd have to make sure I trim the query string so that I can have explicit **case**s, and wouldn't that be less effective?

**Comment by DarkVillager** [comment]
> > Also, maybe use a switch statement instead of a if statement (in combination of the previous comment by Dark)
> 
> I think using if/else is better in this situation because I use the **startsWith** method, and it returns a bool. I'd have to make sure I trim the query string so that I can have explicit **case**s, and wouldn't that be less effective?

or just use a switch statement that check `document.location.pathname.split("/")[1]`
image

**Review by DarkVillager — APPROVED** [review]

File: websites/Y/Yandex/presence.ts [hunk 1]
```diff
@@ -2,25 +2,33 @@ const presence = new Presence({
...
+		case "image": {
+			presenceData.details = `Looking for Images of ${textQuery}`;
+			break;
+		}
+		default:
+			if (currentPath[1] === "video" && currentPath[2] === "search")
+				presenceData.details = `Looking for Videos of ${textQuery}`;
+			else if (currentPath[1] === "news")
+				presenceData.details = `Looking for News on ${textQuery}`;
+			else if (currentPath[1] === "maps") presenceData.details = "Using Maps";
```
Comment (RisingSunLight42) [inline websites/Y/Yandex/presence.ts:31]: Why don't you put them in case ?

**Comment by Nolo001-Aha** [comment]
Yeah sorry, sometimes using "Fix all - Convert..." isn't a good idea. Fixed and checked, should be good to go now (I hope). Sorry for the inconveniences, my first upload here.

**Comment by EncryptedDev** [comment]
Hello there 👋🏼

Thank you for your contribution to the store. But at this current time, we will not proceed with this pull request as I believe it is not at a high enough quality to add to the store. 

There isn't anything wrong with the pull request, but there is just not much to the presence and should add more pages if possible - if this isn't possible because the service doesn't support more, that is fine and let me know so I can reopen the pull request. 

If you need anything, reach out to me on Discord - Encrypted#0001

**Pull request closed without merging**